soc/arm/st_stm32/common/pm_debug_swj.c

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright (c) 2023 STMicroelectronics
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/drivers/pinctrl.h>
+#include <zephyr/init.h>
+
+#define SWJ_NODE DT_NODELABEL(swj_port)
+
+PINCTRL_DT_DEFINE(SWJ_NODE);
+
+const struct pinctrl_dev_config *swj_pcfg = PINCTRL_DT_DEV_CONFIG_GET(SWJ_NODE);
+
+/*
+ * Serial Wire / JTAG port pins are enabled as part of SoC default configuration.
+ * When debug access is not needed and in case power consumption performance is
+ * expected, configure matching pins to analog in order to save power.
+ */
+
+static int swj_to_analog(void)
+{
+	int err;
+
+	/* Set Serial Wire / JTAG port pins to analog mode */
+	err = pinctrl_apply_state(swj_pcfg, PINCTRL_STATE_SLEEP);
```

Comment:
note: I remember now that sleep state requires CONFIG_PM_DEVICE=y, CONFIG_PM alone will not make that state available.

Reply:
I wonder how I got it working then. Wait. I just tested that debug wasn't functional and not measured consumption.. Strange anyway. I'll have a new check.
Then, do you agree to make it available with CONFIG_PM=y only then ? My case, is not limited to PM_DEVICE.